packages/structure/src/main.rs

```diff
@@ -0,0 +1,244 @@
+                "key": format!("{key}"),
+                "contents": inline_content!(contents),
+            })
+            .to_string();
+
+            json.push(json!(format!("<h{level}>")));
+
+            if element == "numbered-heading" {
+                let invocation = format!("[element-number]({key}) ");
+                json.push(inline_content!(invocation));
+            }
+
+            json.push(inline_content!(contents));
+            json.push(json!(format!("</h{level}>")));
+            json.push(json!(
+                {
+                    "name": "list-push",
+                    "arguments": {"name": "structure"},
+                    "data": structure_data,
+                }
+            ));
+
+            print!("{}", serde_json::to_string(&json).unwrap());
+        }
+        other => eprintln!("Cannot convert {element} to {other}!"),
+    }
+}
+
+fn transform_standalone_heading(input: Value, to: &str) {
+    match to {
+        "html" => {
+            let mut json = vec![];
+            let contents = input["data"].as_str().unwrap();
+            let level_arg = input["arguments"]["level"].as_str().unwrap();
+            let level = level_arg.parse::<usize>().unwrap().clamp(1, 6);
+
+            json.push(json!(format!("<h{level}>")));
+            json.push(inline_content!(contents));
+            json.push(json!(format!("</h{level}>")));
+
+            print!("{}", serde_json::to_string(&json).unwrap());
+        }
+        other => eprintln!("Cannot convert standalone-heading to {other}!"),
+    }
+}
+
+fn transform_toc(input: Value, to: &str) {
+    let max_level = input["arguments"]["max-level"].as_u64().unwrap() as usize;
+    let toc = TOC::build_from_list(max_level);
+
+    match to {
+        "html" => print!("{}", toc.to_html()),
+        other => eprintln!("Cannot convert table-of-contents to {other}!"),
+    }
+}
+
+fn transform(from: &str, to: &str) {
+    let input: Value = {
+        let mut buffer = String::new();
+        io::stdin().read_to_string(&mut buffer).unwrap();
+        from_str(&buffer).unwrap()
+    };
+
+    match from {
+        "table-of-contents" => transform_toc(input, to),
+        "unnumbered-heading" => transform_heading(input, to, from),
+        "numbered-heading" => transform_heading(input, to, from),
+        "standalone-heading" => transform_standalone_heading(input, to),
+        "element-number" => transform_element_number(input),
+        "label-to-key" => transform_label_to_key(input),
+        other => {
+            eprintln!("Package does not support {other}");
+        }
+    }
+}
+
+fn manifest() {
+    print!(
+        "{}",
+        json!({
+            "name": "structure",
+            "version": "0.1",
+            "description": "This package manages document structure to provide numbering for headings, figures and tables.",
+            "transforms": [
+                {
+                    "from": "table-of-contents",
+                    "to": ["any"],
```

Comment:
Would be really nice to support all formats similiar to how the bibliography module works by generating other ModMark modules like link etc. instead of handling the low-level transformation.

But since we on line 124 will produce a error if the format is not HTML I think we should say just "html" in the manifest. Or add more general support (at least for LaTeX if we feel like we have time). 

```suggestion
                    "to": ["html"],
```